diff --git a/Contrib/PhantomJS/HTMLCS_Run.js b/Contrib/PhantomJS/HTMLCS_Run.js
index 3badef6..8c1b091 100755
--- a/Contrib/PhantomJS/HTMLCS_Run.js
+++ b/Contrib/PhantomJS/HTMLCS_Run.js
@@ -90,8 +90,8 @@ if (system.args.length < 3 || system.args.length > 4) {
         }
 
         console.log('');
-        console.log(_global.HTMLCS.getTranslation("auditor_errors") + ': ' + messages['ERROR'].length + ', ' + _global.HTMLCS.getTranslation("auditor_warnings") + ': ' + messages['WARNING'].length +
-            ', ' + _global.HTMLCS.getTranslation("auditor_notices") + ': ' + messages['NOTICE'].length);
+        console.log('Errors: ' + messages['ERROR'].length + ', Warnings: ' + messages['WARNING'].length +
+            ', Notices: ' + messages['NOTICE'].length);
         cb();
     }
 
